src/pyqasm/modules/base.py

```diff
@@ -525,6 +527,209 @@ def unroll(self, **kwargs):
+        """Rebase the AST to use a specified target basis set.
+        
+        Will unroll the module if not already done.
+
+        Args:
+            target_basis_set: The target basis set to rebase the module to.
+            in_place (bool): Flag to indicate if the reversal should be done in place.
+            
+        Returns:
+            QasmModule: The module with the gates rebased to the target basis set.
+        """
+        if target_basis_set not in DECOMPOSITION_RULES:
+            raise ValueError(f"Target basis set '{target_basis_set}' is not defined.")
+
+        qasm_module = self if in_place else self.copy()
+
+        if len(qasm_module._unrolled_ast.statements) == 0:
+            qasm_module.unroll()
+
+        rebased_statements = []
+
+        for statement in qasm_module._unrolled_ast.statements:
+            if isinstance(statement, QuantumGate):
+                gate_name = statement.name.name
+
+                # Decompose the gate
+                processed_gates_list = self._process_gate_statement(
+                    gate_name,
+                    statement,
+                    target_basis_set
+                    )
+                for processed_gate in processed_gates_list:
+                    rebased_statements.append(processed_gate)
+
+            elif isinstance(statement, BranchingStatement):
+                # Recursively process the if_block and else_block
+
+                rebased_statements.append(self._process_branching_statement(
+                    statement,
+                    target_basis_set
+                    ))
+
+            else:
+                # Non-gate statements are directly appended
+                rebased_statements.append(statement)
+
+        # Replace the unrolled AST with the rebased one
+        qasm_module._unrolled_ast.statements = rebased_statements
+
+        return qasm_module
+
+    def _process_gate_statement(self, gate_name, statement, target_basis_set):
+        """Process the gate statement based on the target basis set.
+        
+        Args:
+            gate_name: The name of the gate to process.
+            statement: The statement to process.
+            target_basis_set: The target basis set to rebase the module to.
+            
+        Returns:
+            list: The processed gates based on the target basis set.
+        """
+        decomposition_rules = DECOMPOSITION_RULES[target_basis_set]
+        target_basis_gate_list = BASIS_GATE_MAP[target_basis_set]
+
+        processed_gates_list = []
+
+        if gate_name in target_basis_gate_list:
+            # Keep the gate as is
+            processed_gates_list = [statement]
+        elif gate_name in decomposition_rules:
+            # Decompose the gates
+            processed_gates_list = self._get_decomposed_gates(
+                decomposition_rules,
+                statement,
+                gate_name
+                )
+        elif self._is_parameterized_gate(gate_name):
```

Comment:
```suggestion
        elif gate_name in {"rx", "ry", "rz"}:
```
Let's keep it simple , we don't need to make a helper method for this check